hscontrol/mapper/mapper.go

```diff
@@ -257,11 +259,6 @@ func (m *Mapper) PeerChangedResponse(
 ) ([]byte, error) {
 	resp := m.baseMapResponse()
 
-	peers, err := m.ListPeers(node.ID)
```

Comment:
I agree this is not necessary, but instead of listing all, we should just list the one from the change map.

It would make more sense to make `ListPeers(node NodeID, peerIDs ...NodeID)` and only fetch the ones that are needed here.

Reply:
I opted against changing `ListPeers` for now. It is also used for full map responses, where I think all peers are needed (although I have not looked too deeply into this function yet). But I can of course change the naming scheme if you like something like `ListPeersSubset` better than `ListNodesSubset`.

Reply:
I think modifying both `ListNodes` and `ListPeers` to be able to take a optional list arg would be the best.